### ISSUE TITLE
Use of multiple accounts when rate limited [New feature

### ISSUE THREAD

**Issue by slarrain** [DocNow/twarc#245]

**Comment by slarrain** [comment]
I've been thinking of a way to extract multiple tweets and accounts and bypass (sort of) the rate limit imposed by Twitter.

Here is what I thought:

What if we have multiple accounts configured on twarc, and the moment we get rate limited, we go to the next account in our `.twarc` file and so on, until we reached our final account (or the original) and then we actually wait until the timeout is over. 
Obviously, this would be an optional feature for the user and not by default. But I think that, at least in my case, it would be very useful. But I'm not sure if it would be useful for anybody else.

I've even been looking into how to implement it, and I think the best way would be on the decorator function, that actually waits (sleeps) for the timeout.

What do the rest of the community think? Is it worth implementing it?

**Comment by slarrain** [comment]
That is indeed problematic. I have not seen that part of the Twitter Developer Policy and Agreement. Thanks for pointing it out.

**Comment by edsu** [comment]
I think it definitely would be problematic re the TOS to register multiple app keys. I think it also could be weird if multiple accounts are registered by one person as a type of sock puppet to get multiple access keys. One thing I’ve been curious about in the case of [DocNow](https://github.com/docnow/DocNow) is whether a group of actual people could work collaboratively to document an event, and contribute their keys to the effort.

**Comment by slarrain** [comment]
Oh, I was always thinking of multiple keys from different people. Which, reading the TOS again, could actually be acceptable. One person would not be circumventing the rate limit. But instead, multiple people would be using their own keys, which in theory, should be acceptable. And if all of them reach the limit, then they would have to wait.
Is actually a way of having a group of people make the same search together.

**Comment by edsu** [comment]
twarc itself doesn't really offer any functionality for group collaboration -- but this is something we are thinking about enabling in the [docnow](https://github.com/docnow/docnow) application. if you have any thoughts about how that could work, please add them to that issue tracker!

**Comment by SamHames** [comment]
Yeah - I should have been a little bit more nuanced because I think in terms of complying with the terms of service it matters where those keys came from, and how they're being used. 

With multiple people contributing keys (I'm assuming doing the OAuth dance to collect user credentials?) the tricky parts in the terms of service are about how the data is shared. Something I'm trying to get my head around in the research context/institutional ethics frameworks right now.

The other thing to keep in mind is that the recent changes to the developer signup process mean that it's not going to get any easier to get application credentials since Twitter actually seems to be reviewing them more closely now.

**Comment by igorbrigadir** [comment]
I think we should revisit this idea, not so much for multiple accounts but to make sure we take advantage of endpoints that have "app only" limits and access token limits, like the followers ones that are super restrictive.
